content/en/platform/corda/5.0/developing-applications/api/ledger/utxo-ledger/utxo-advanced-ledger-extensions/ledger-extensions-api-reference.md

```diff
@@ -351,6 +369,45 @@ public final class ExampleIdentifiableContract extends IdentifiableContract {
 }
 ```
 
+#### Creating, Updating and Deleting Identifiable States
+
+To create a unique identifiable state, create an instance of an `IdentifiableState` with an `id` of `null`.
+
+To update an existing identifiable state, do the following:
+
+1. Retrieve the existing `IdentifiableState`, `id` or its `StateRef` if the `id` is `null`.
```

Comment:
I don't think this matches up with what i wrote:

> - Retrieve the existing `IdentifiableState`'s `id` or its `StateRef` if the `id` is `null`.

The main point is that you're extracting the `id` or the `StateRef` from the state once you've retrieved it, but you only use the `StateRef` if the `id` is `null` which happens when its the first time it is updated. There might be a better way of wording all that.